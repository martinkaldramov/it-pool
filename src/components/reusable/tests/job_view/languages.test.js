import React from 'react';
import { shallow } from 'enzyme';
import Languages from '../../job_view/languages';

describe('Languages', () => {
  const mockLanguages = ['English', 'French', 'German'];
  const wrapper = shallow(<Languages languages={mockLanguages} />);

  it('should render successfully', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render all the passed languages', () => {
    expect(wrapper.find('ul').children().length).toBe(mockLanguages.length);
  });
});
