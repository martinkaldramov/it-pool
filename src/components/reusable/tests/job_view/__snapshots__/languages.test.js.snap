// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Languages should render successfully 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Languages
    languages={
        Array [
            "English",
            "French",
            "German",
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <h4>
          LANGUAGES
</h4>,
        <ul>
          <li>
                    English
          </li>
          <li>
                    French
          </li>
          <li>
                    German
          </li>
</ul>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "LANGUAGES",
        },
        "ref": null,
        "rendered": "LANGUAGES",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              English
</li>,
            <li>
              French
</li>,
            <li>
              German
</li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "English",
            },
            "ref": null,
            "rendered": "English",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "French",
            },
            "ref": null,
            "rendered": "French",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "German",
            },
            "ref": null,
            "rendered": "German",
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <h4>
            LANGUAGES
</h4>,
          <ul>
            <li>
                        English
            </li>
            <li>
                        French
            </li>
            <li>
                        German
            </li>
</ul>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "LANGUAGES",
          },
          "ref": null,
          "rendered": "LANGUAGES",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                English
</li>,
              <li>
                French
</li>,
              <li>
                German
</li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "English",
              },
              "ref": null,
              "rendered": "English",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "French",
              },
              "ref": null,
              "rendered": "French",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "German",
              },
              "ref": null,
              "rendered": "German",
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
